src/main.rs

```diff
@@ -1,28 +1,63 @@
+use std::io::{Stdin, Stdout, Write};
+
 fn main() {
     println!("Olá! 😃");
 
-    loop {        
-        println!("\nVocê gostaria de adicionar um novo TODO? 🤔 (Digite: 's' para SIM ou qualquer outra tecla para NÃO)");
+    loop {
+       let mut ask_todo = Terminal::new();
+       let todo = ask_todo.ask_for_new_todo();
 
-        if input() == "s" {
-            create_todo();
-        } else {           
-            println!("\nTodo list finalizado! 🤠");
-            break;
-        }
+       ask_todo.show_todo(&todo);
     }
 }
 
-fn create_todo() {
-    
-    println!("\nQual TODO deseja criar?");
-    let todo = input();
+#[derive(Debug, Clone)]
+struct Todo {
+    message: String
+}
 
-    println!("\n✅: {todo}");
+struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
 }
 
-fn input() -> String {
-    let mut buf = String::new();
-    std::io::stdin().read_line(&mut buf).unwrap();
-    buf.trim().to_string()
- }
+impl Terminal {
+    fn new() -> Self {
+
+        Terminal {
+            stdin: std::io::stdin(), 
+            stdout: std::io::stdout() 
+        }
+    }
+
+    fn should_ask_for_todo(&mut self) -> bool {
+        let mut buf = String::new();
+        self.stdin.read_line(&mut buf).unwrap();
+
+        buf.trim() == "s"
+    }
+
+    fn ask_for_new_todo(&mut self) -> Todo {          
+
+        println!("\nVocê gostaria de adicionar um novo TODO? 🤔 (Digite: 's' para SIM ou qualquer outra tecla para NÃO)");
+        
+        let response = self.should_ask_for_todo();
+        
+        if !response {
```

Comment:
Como só se usa a variável uma vez, eu já colocaria direto no `if`.

```suggestion
        if !self.should_ask_for_todo() {
```

Fora isso, acho que tá ótimo! Já vou deixar aprovado.

Reply:
Elimina a declaração de uma variável, ok! Obrigado! :D